### ISSUE TITLE
buildx build is broken on armv7 for alpine:edge

### ISSUE THREAD

**Issue by doanac** [docker/buildx#508]
I've hit an interesting issue trying to build alpine:edge on armv7. I'm seeing things in alpine about [time64](https://musl.libc.org/time64.html) and it seems moby has done some seccomp changes to deal with this recently. I'm running docker 20.10.2 and buildx 0.5.1 (docker info below) on an [apalis imx6](https://www.toradex.com/computer-on-modules/apalis-arm-family/nxp-freescale-imx-6).

## Steps to reproduce
~~~
wget -O ~/.docker/cli-plugins/docker-buildx https://github.com/docker/buildx/releases/download/v0.5.1/buildx-v0.5.1.linux-arm
chmod +x  ~/.docker/cli-plugins/docker-buildx
docker buildx create --driver-opt image=moby/buildkit:master --use
cat >Dockerfile <<EOF
FROM alpine:edge
RUN apk --update add       libstdc++       libmicrohttpd
EOF
DOCKER_CLI_EXPERIMENTAL=enabled docker buildx build .
~~~

The failed out is
~~~
WARN[0000] No output specified for docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load 
[+] Building 58.8s (8/8) FINISHED                                                                              
 => [internal] booting buildkit                                                                          23.3s
 => => pulling image moby/buildkit:master                                                                 3.8s
 => => creating container buildx_buildkit_vibrant_mclean0                                                19.5s
 => [internal] load build definition from Dockerfile                                                      0.5s
 => => transferring dockerfile: 145B                                                                      0.1s
 => [internal] load .dockerignore                                                                         0.8s
 => => transferring context: 2B                                                                           0.0s
 => resolve image config for docker.io/docker/dockerfile:1.2.1                                            4.7s
 => docker-image://docker.io/docker/dockerfile:1.2.1@sha256:e2a8561e419ab1ba6b2fe6cbdf49fd92b95912df1cf7  8.3s
 => => resolve docker.io/docker/dockerfile:1.2.1@sha256:e2a8561e419ab1ba6b2fe6cbdf49fd92b95912df1cf7d313  0.1s
 => => sha256:2cf4d8260a9d61b7e02bf54781629b81d0b9bfc6cd166d424b9feeab8c9df815 9.01MB / 9.01MB            5.1s
 => => extracting sha256:2cf4d8260a9d61b7e02bf54781629b81d0b9bfc6cd166d424b9feeab8c9df815                 3.1s
 => [internal] load metadata for docker.io/library/alpine:edge                                            2.6s
 => [1/2] FROM docker.io/library/alpine:edge@sha256:56e8a54633573bc40a003884a01e60fb31d070d05e220d244e88  4.4s
 => => resolve docker.io/library/alpine:edge@sha256:56e8a54633573bc40a003884a01e60fb31d070d05e220d244e88  0.1s
 => => sha256:b0bb8106b6fc46e77f4535e17ad4a0ec520bcac32791bd2633e9da14c72dd7b9 2.42MB / 2.42MB            2.7s
 => => extracting sha256:b0bb8106b6fc46e77f4535e17ad4a0ec520bcac32791bd2633e9da14c72dd7b9                 1.6s
 => ERROR [2/2] RUN apk --update add       libstdc++       libmicrohttpd                                  2.9s
------                                                                                                         
 > [2/2] RUN apk --update add       libstdc++       libmicrohttpd:                                             
#8 0.685 fetch https://dl-cdn.alpinelinux.org/alpine/edge/main/armv7/APKINDEX.tar.gz                           
#8 0.722 ERROR: https://dl-cdn.alpinelinux.org/alpine/edge/main: temporary error (try again later)             
#8 0.723 fetch https://dl-cdn.alpinelinux.org/alpine/edge/community/armv7/APKINDEX.tar.gz                      
#8 0.724 WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/edge/main: No such file or directory          
#8 0.740 ERROR: https://dl-cdn.alpinelinux.org/alpine/edge/community: temporary error (try again later)
#8 0.741 WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/edge/community: No such file or directory
#8 0.752 ERROR: unable to select packages:
#8 0.769   libmicrohttpd (no such package):
#8 0.769     required by: world[libmicrohttpd]
#8 0.769   libstdc++ (no such package):
#8 0.769     required by: world[libstdc++]
------
Dockerfile:4
--------------------
   2 |     FROM alpine:edge
   3 |     
   4 | >>> RUN apk --update add       libstdc++       libmicrohttpd
   5 |     
--------------------
error: failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c apk --update add       libstdc++       libmicrohttpd]: exit code: 2
~~~

## docker info
~~~
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., v0.5.1)

Server:
 Containers: 5
  Running: 5
  Paused: 0
  Stopped: 0
 Images: 8
 Server Version: 20.10.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: journald
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 269548fa27e0089a8b8278fc4fc781d7f65a939b.m
 runc version: 49a73463331bd8ff44bb8349e33f4b2e1ae34b4f-dirty
 init version: de40ad0-dirty (expected: de40ad007797e)
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.5-lmp-standard
 Operating System: Linux-microPlatform 3.2.1-1164-77-18-g34e6bbd
 OSType: linux
 Architecture: armv7l
 CPUs: 4
 Total Memory: 965.3MiB
 Name: apalis-imx6-1
 ID: 7FTP:PAYH:BIO3:RXOF:VEOS:VUNS:PL6D:4OW6:NMDQ:LHQW:3T6L:JW6B
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio weight support
WARNING: No blkio weight_device support
~~~

This does work under normal `docker build`.

I have a slightly more convoluted way of reproducing this on an AWS Gravitron as well.

**Comment by jeptechnology** [comment]
We are also experiencing this issue using the newly released `alpine:3.13`:
```
|  #11 0.054 fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/armv7/APKINDEX.tar.gz
|  #11 0.092 4155077520:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
|  #11 0.092 4155077520:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
|  #11 0.092 4155077520:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
|  #11 0.092 4155077520:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
|  #11 0.092 4155077520:error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1913:
|  #11 0.094 ERROR: https://dl-cdn.alpinelinux.org/alpine/v3.13/main: Permission denied
|  #11 0.094 WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.13/main: No such file or directory
|  #11 0.094 fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/armv7/APKINDEX.tar.gz
|  #11 0.131 4155077520:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
|  #11 0.131 4155077520:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
|  #11 0.131 4155077520:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
|  #11 0.131 4155077520:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
|  #11 0.131 4155077520:error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1913:
|  #11 0.132 ERROR: https://dl-cdn.alpinelinux.org/alpine/v3.13/community: Permission denied
|  #11 0.132 WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.13/community: No such file or directory

```

**Comment by doanac** [comment]
I have some more debug on this. Its taken a bit to get past all the indirection. I'm recreating this purely in moby/buildkit now. I run a buildkitd daemon in one terminal and run `buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=../` in the other.

The problem seems to be the buildkitd daemon. Its built against alpine 3.12. It gets more fun than that, but I'll create a moby/buildkit bug for this now. I'm 95% sure there's nothing in docker/buildx that needs fixing.

Here's the bug for cross-reference - https://github.com/moby/buildkit/issues/1946
